src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -101,10 +98,29 @@ private void CheckReadWriteArguments(byte[] array, int offset, int count)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNumRequired);
             if (array.Length - offset < count)
                 throw new ArgumentException(SR.Argument_InvalidOffLen);
+        }
+
+        private void CheckReadWriteArguments()
+        {
             if (_handle == null)
                 InternalResources.FileNotOpen();
         }
 
+        private void CheckReadWriteArguments(byte[] array, int offset, int count)
+        {
+            CheckArrayArguments(array, offset, count);
+
+            CheckReadWriteArguments();
+        }
+
+        private void CheckWriteArguments()
+        {
+            if (_inBreak)
+                throw new InvalidOperationException(SR.In_Break_State);
+
+            CheckReadWriteArguments();
```

Comment:
it's surprising to see `CheckWriteArguments` calling `CheckReadWriteArguments` (I would expect the first to be the subset of the latter). Perhaps `CheckReadWriteArguments` could be renamed to sth like `CheckHandle` or `EnsureOpened`?

Reply:
These mimic the existing pattern.. should I change the existing overloads?

Reply:
I think it's fine to just rename `CheckReadWriteArguments()` to `CheckHandle()`